client/my-sites/stats/stats-date-control/stats-date-control-picker.tsx

```diff
@@ -7,18 +7,67 @@ import DateControlPickerShortcuts from './stats-date-control-picker-shortcuts';
 import { DateControlPickerProps } from './types';
 
 const DateControlPicker = ( { slug, queryParams }: DateControlPickerProps ) => {
+	const redirectToQueryDate = ( daysToSubtract: number ) => {
+		const startDate = calculateQueryDate( daysToSubtract );
+		const query = qs.stringify( Object.assign( {}, queryParams, { startDate } ), {
+			addQueryPrefix: true,
+		} );
+		const period = 'day'; // TODO: adjust this as needed
+		const url = `/stats/${ period }/${ slug }`;
+		const href = `${ url }${ query }`;
+
+		page( href );
+	};
+
 	// shortcut list will come from props
 	const shortcutList = [
 		{
-			id: 'test',
-			label: 'Test shortcut',
-			onClick: () => {
-				// eslint-disable-next-line no-console
-				console.log( 'clicking' );
-			},
+			id: 'today',
+			label: 'Today',
+			onClick: () => redirectToQueryDate( 0 ),
```

Comment:
I think we should avoid using a function pointer here and just past the offsets. The way I understand it, selecting a shortcut updates the start and end input fields so we should probably provide an offset and a range. As an example, Today might look like this:

```
{
  id: 'today',
  label: 'Today',
  offset: 0, // offset from current date
  range: 30, // as in 30 days
}
```

Then we could pass a generic callback to the `DateControlPickerShortcuts` component that handled updating the input fields based on the offset/range values. Something like this:

```
  <DateControlPickerShortcuts shortcutList={ shortcutList, handleShortcutApplied } />
```

Then we'd do the updating of fields in that callback.

```
const handleShortcutApplied = ( daysOffset: number, range: number ) => {
	const newStartDate = calculateQueryDate( daysOffset );
	changeStartDate( newStartDate );
	const newEndDate = 0; // calc the necessary date value
	changeEndDate( newEndDate );
}
```

This way we avoid breaking the encapsulation of the control (via the function pointers in `props.shortcuts`) and only update the caller when the Apply button is clicked.